04/SpeedTest.js

```diff
@@ -1,17 +1,40 @@
-import React, { useEffect, useRef, useState } from 'react';
+import React, { useEffect, useRef, useState, useReducer } from 'react';
 import useRandomItem from './hook';
+import './style.css';
+import Table from './table';
 
 const SpeedTest = () => {
-    const [word, regenerateWord] = useRandomItem(['devmentor.pl', 'abc', 'JavaScript']);
+    const [generatedWord, regenerateWord] = useRandomItem(['devmentor.pl', 'abc', 'JavaScript']);
 
     useEffect(() => {
         regenerateWord();
     }, []);
 
     const [text, setText] = useState('');
-    const [time, setTime] = useState(0);
+    const [taskTime, setTime] = useState(0);
     const [timeArray, setArray] = useState([]);
+
+    function reducer(state, action) {
+        switch (action.type) {
+            case 'word':
+                return { word: text, time: 0, init: true };
+            case 'time':
+                setText('');
+                return {
+                    word: state.word,
+                    time: timeArray[timeArray.length - 1],
+                    init: false,
+                };
+            default:
+                throw new Error();
+        }
+    }
+
     const [signs, setSigns] = useState(0);
```

Comment:
Można też tak: `const [, setSigns] = useState(0);` 
Swoją drogą to czy jest to w ogóle potrzebne jeśli nie odczytujesz tej wartości?